components/tracker/src/lib.rs

```diff
@@ -69,6 +69,11 @@ impl Tracker {
         detail.set_apply_write_wal_nanos(self.metrics.apply_wait_nanos);
         detail.set_apply_write_memtable_nanos(self.metrics.apply_write_memtable_nanos);
     }
+
+    pub fn write_time_detail(&self, detail: &mut pb::TimeDetail) {
+        detail.set_wait_wall_time_ms(self.metrics.wait_wall_time_ms);
+        detail.set_kv_read_wall_time_ms(self.metrics.kv_read_wall_time_ms);
```

Comment:
Perhaps these durations can be broken down better. 

I'm going to further add these durations about the duration before `async_write`: https://github.com/pingcap/kvproto/pull/965

Reply:
Got it, I'm thinking about the dimension to split the durations, one way is to separate them by the `read` and `write` path just like the fast-tuning does.

Reply:
I add the wait time in the waiter manager in https://github.com/pingcap/kvproto/pull/965 as part of `WriteDetail`. The newly added fields are specific to write commands, which might extend the original semantics of `WriteDetail` (which only includes the raftstore stuff).
Do you think it a good place to put the wait time of pessimistic locks?

Reply:
How about splitting a command into `read` and `write` parts, the `read` part is the processing before async write which may include things like conflict check, existence check, value fetch, assertion checks, etc. All the durations in the async write are put into the `write` part. A possible way is:
- The `TimeDetail` could be used to represent the `read` part which coprocessor dose in the before.
- The `WriteDetail` could be used to represent the `write` part, and it includes raftstore metrics already.

Them the pessimistic lock wait duration could be put into the `TimeDetail` as it's in the `read` part.

Reply:
@you06 @MyonKeminta 
Any suggestions about how to represent different durations of different stags in the [proto](https://github.com/pingcap/kvproto/pull/965)?

Reply:
Personally I prefer considering things like pessimistic lock waiting and latch waiting as part of writing instead of reading. But since `TimeDetail` doesn't seem to intend to representing stuff related to reading only, I think I don't have any strong reason to prefer one to another.

Reply:
Looks `TimeDetail` is the general time of both read and write requests, and the lock related durations are more close to write requests, suppose there may be the write-push-down feature in the future TiKV, I prefer to place them in `WriteDetail`.

Reply:
Another option is to keep the meaning of `WriteDetail` to represent raftstore metrics, and put other durations into `TimeDetail`, so the `WriteDetail` would not become too large or complex.